fileinfo.py

```diff
@@ -50,24 +56,31 @@
     't_char': t_char
 }
 
-print("\nfile name =", file_info['fname'])
-print("file size =", file_info['fsize'], "bytes")
-print("last modified =", file_info['f_lm'])
-print("last accessed =", file_info['f_la'])
-print("creation time =", file_info['f_ct'])
-print("Total number of lines are =", file_info['no_of_lines'])
-print("Total number of characters are =", file_info['t_char'])
+file_info_fmt = ('\nfile name = {}\nfile size = {}\nlast modified = {}\n'
+                 'last accessed = {}\ncreation time = {}\n'
+                 'Total number of lines are = {}\n'
+                 'Total number of characters are = {}')
+print(
+    file_info_fmt.format(file_info['fname'],
+                         str(file_info['fsize']) + " bytes", file_info['f_lm'],
+                         file_info['f_la'], file_info['f_ct'],
+                         file_info['no_of_lines'], file_info['t_char']))
 
 if stat.S_ISDIR(file_stats[stat.ST_MODE]):
     print("This a directory")
 else:
-    print("This is not a directory\n")
-    print("A closer look at the os.stat(%s) tuple:" % file_name)
-    print(file_stats)
-    print("\nThe above tuple has the following sequence:   ")
-    print("""st_mode (protection bits), st_ino (inode number), 
-    st_dev (device),    st_nlink (number of hard links),    
-    st_uid (user ID of owner),   st_gid (group ID of owner),    
-    st_size (file size, bytes),  st_atime (last access time, seconds since epoch),  
-    st_mtime (last modification time),   st_ctime (time of creation, Windows)"""
-          )
+    file_stats_fmt = ''
+    print("\nThis is not a directory.")
+    stats_strings = ("st_mode (protection bits),"
+                     "st_ino (inode number),"
+                     "st_dev (device),"
+                     "st_nlink (number of hard links),"
+                     "st_uid (user ID of owner),"
+                     "st_gid (group ID of owner),"
+                     "st_size (file size bytes),"
+                     "st_atime (last access time seconds since epoch),"
+                     "st_mtime (last modification time),"
+                     "st_ctime (time of creation Windows)")
+    stats_keys = stats_strings.split(',')
+    for key, value in OrderedDict(zip(stats_keys, file_stats)).items():
```

Comment:
You can leave out OrderedDict and .items, as iterating through key and value of the zipped pairs will be sufficient.

Reply:
Great! It could make the code more concise, thanks your opinion.

Reply:
Great! It could make the code more concise, thanks your opinion.

Reply:
Cool, thanks for updating.